### PR TITLE
Site Editor Domain Upsell: Add callout (using editing toolkit)

### LINKED ISSUE #73445: Fix zIndex at the new domain upsell on editor
Slack thread: p1676569166916009-slack-CKZHG0QCR

Even with zIndex = 1 it still stays in front because it’s outside of the iframe. We’ll need to investigate more how to fix it:

Image

### LINKED ISSUE #76321: Editor domain upsell: Gray out the upsell behind the modal
See p1682496634800579-slack-CKZHG0QCR and p9Jlb4-6AL-p2#general-write

The domain upsell at the top of the editor isn't being grayed out like the editor, when the modal overlay is visible. 
Also, the z-index may be too high. See p1682515858354489/1682446289.812659-slack-C9EJ7KSGH and p9Jlb4-6AL-p2

Acceptance criteria:
- [ ] The domain upsell is grayed out like the editor, when the modal overlay is visible.
- [ ] The upsell has the correct z-index.

Screenshots:

image
image

### PR DESCRIPTION
**Pull request by lupus2k**
Fixes https://github.com/Automattic/wp-calypso/issues/76321
Fixes https://github.com/Automattic/wp-calypso/issues/73445

## Proposed Changes

Add Domain Upsell callout to Gutenberg using [editing toolkit](https://github.com/Automattic/wp-calypso/blob/trunk/apps/editing-toolkit/README.md)

#### Conditions to show the callout
- User has a verified email
- Site has no custom domain
- Site is not a P2

#### Track events
- View: `calypso_block_editor_domain_upsell_callout_view`
- Click: `calypso_block_editor_domain_upsell_callout_click`
- Dismiss: `calypso_block_editor_domain_upsell_callout_dismiss`

## Testing Instructions

- You have to sandbox `widgets.wp.com`
- Go to `/apps/editing-toolkit`
- Run `yarn dev --sync`
- On Calypso, go to `/posts` (and `/pages`) and edit or create new.
- Or, change the view to classic and use `/wp-admin`
- You should see the callout if conditions matches.
- Check CTA
- Check dismiss (you can re-enable it by running `wp.data.dispatch( 'core/preferences' ).set( 'core/edit-post', 'domain_upsell_callout_dismiss', false );` in the console and reloading the page.
- Check track events

To test on Atomic Sites, follow the instructions here: PCYsg-ly5-p2 (**ETK and Atomic Sites**)

## Screenshots
image
image
image
image
image

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] ~~Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)~~
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] ~~For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?~~

### CODE REVIEW

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-domain-upsell-callout/src/domain-upsell-callout.jsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+			</div>
+		</>
+	);
+};
+
+async function showDomainUpsellCallout() {
+	// We wait to the editor to be ready because showing an upsell before the editor could be shocking
+	await isEditorReady();
+	if ( shouldShowDomainUpsell() ) {
+		const toolbarContainer = document.querySelector( '.edit-post-header-toolbar' );
```
Comment (simison) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-domain-upsell-callout/src/domain-upsell-callout.jsx:103]: You should bail out if the element isn't found.

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-domain-upsell-callout/src/domain-upsell-callout.jsx [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { dispatch, select, subscribe } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { close, globe, Icon } from '@wordpress/icons';
+import { createElement } from 'react';
```
Comment (simison) [inline apps/editing-toolkit/editing-toolkit-plugin/wpcom-domain-upsell-callout/src/domain-upsell-callout.jsx]: Not a blocker but see also https://github.com/WordPress/gutenberg/tree/trunk/packages/element#readme

**Comment by simison** [comment]
Note also accounting for distraction free editing — and this can be a follow up, too:

image

**Comment by lupus2k** [comment]
Thanks for your suggestions @simison.

Do we have a function on Atomic sites to know if the site has custom domain names? (like [this](https://github.com/Automattic/wp-calypso/pull/76855/files#diff-0b0c7460d276c8e2f9534f6c181e79cc5865b09f3c92090a26e0ba0d4b731945R137-R143) for simple sites). @candy02058912 found a solution but I'm double checking we are not duplicating an existing function

File: apps/editing-toolkit/editing-toolkit-plugin/wpcom-domain-upsell-callout/src/domain-upsell-callout.jsx (hunk 2 above)
Comment (lupus2k) [reply apps/editing-toolkit/editing-toolkit-plugin/wpcom-domain-upsell-callout/src/domain-upsell-callout.jsx]: Done ✅

**Comment by lupus2k** [comment]
> Note also accounting for distraction-free editing — and this can be a follow up, too:

Hey, @simison, I tried to reproduce it, but in distraction-free mode, I see it in the toolbar, not outside of it. Can you show me how you do this?

image

**Comment by lupus2k** [comment]
Btw, I found it disappeared when entering in distraction-free, but reloading, I see it on the toolbar but moving a little weird when showing the toolbar.


https://github.com/Automattic/wp-calypso/assets/402286/f6d69c19-a853-4130-9be2-daa80a58b0ea

**Comment by lupus2k** [comment]
I created [this follow-up](https://github.com/Automattic/dotcom-forge/issues/2445) for the `Distraction free` issue

**Review by candy02058912 — COMMENTED** [review]
double upsell

Do we want to remove [this](https://github.com/Automattic/wp-calypso/blob/test/domain-upsell-callout-etk/client/components/web-preview/content.jsx#L395) or [this](https://github.com/Automattic/wp-calypso/blob/test/domain-upsell-callout-etk/client/gutenberg/editor/calypsoify-iframe.tsx#L723) in this PR (didn't look too deep yet), so it doesn't show a double upsell like the above screenshot?
It was showing for one of my free plan sites.

The rest is working nicely; thanks for working on this.

**Comment by lupus2k** [comment]
> Do we want to remove [this](https://github.com/Automattic/wp-calypso/blob/test/domain-upsell-callout-etk/client/components/web-preview/content.jsx#L395) or [this](https://github.com/Automattic/wp-calypso/blob/test/domain-upsell-callout-etk/client/gutenberg/editor/calypsoify-iframe.tsx#L723) in this PR (didn't look too deep yet), so it doesn't show a double upsell like the above screenshot?

Thanks for noticing this. I removed from `calypsoify-iframe` because this is for the editor, [this one](https://github.com/Automattic/wp-calypso/blob/test/domain-upsell-callout-etk/client/components/web-preview/content.jsx#L395) is for the preview and will continue showing there.

**Review by paulopmt1 — APPROVED** [review]
It works as expected Atomic and Simple sites, good job @lupus2k!

**Pull request merged**